src/index.js

```diff
@@ -22,35 +38,37 @@ function decode(res) {
   return JSON.parse(jsonPayload);
 }
 
-function backwardCompat(node, attr, key) {
-  return `${node}[data-mystique-key="${key}"],${node}[data-${attr}="${key}"]`;
+function toScript(attr, key, props, id) {
+  return `<script type="application/json" data-${attr}="${key}" data-hypernova-id=${id}>${LEFT}${encode(props)}${RIGHT}</script>`; // eslint-disable-line max-len
 }
 
-function toScript(attr, key, props) {
-  return `<script type="application/json" data-${attr}="${key}">${LEFT}${encode(props)}${RIGHT}</script>`; // eslint-disable-line max-len
-}
-
-function fromScript(attr, key) {
-  const node = document.querySelector(backwardCompat('script', attr, key));
+function fromScript(attr, key, id) {
+  const idSelector = id ? `[data-hypernova-id="${id}"]` : '';
+  const node = document.querySelector(`script[data-${attr}="${key}"]${idSelector}`);
   if (!node) return null;
   const jsonPayload = node.innerHTML;
+
   return decode(jsonPayload.slice(LEFT.length, jsonPayload.length - RIGHT.length));
 }
 
 function serialize(name, html, data) {
   const key = name.replace(/\W/g, '');
-  const markup = `<div data-hypernova-key="${key}">${html}</div>`;
-  const script = toScript('hypernova-key', key, data);
+  const id = uuid();
+  const markup = `<div data-${DATA_KEY}="${key}" data-hypernova-id=${id}>${html}</div>`;
+  const script = toScript(DATA_KEY, key, data, id);
   return `${markup}\n${script}`;
 }
 
 function load(name) {
   const key = name.replace(/\W/g, '');
-  const node = document.querySelector(backwardCompat('div', 'hypernova-key', key));
-  if (!node) return {};
+  const nodes = document.querySelectorAll(`div[data-${DATA_KEY}="${key}"]`);
+  if (!nodes.length) return [];
```

Comment:
fwiw this isn't actually necessary at this point, since `map` will do the short-circuit for us